src/components/Trade/hooks/useSwapper/swapperManager.ts

```diff
@@ -114,41 +112,49 @@ export const getSwapperManager = async (flags: FeatureFlags): Promise<SwapperMan
       web3: polygonWeb3,
       adapter: polygonChainAdatper,
     })
-    _swapperManager.addSwapper(zrxPolygonSwapper)
+    swapperManager.addSwapper(zrxPolygonSwapper)
   }
 
   if (flags.ThorSwap) {
-    await (async () => {
-      const midgardUrl = getConfig().REACT_APP_MIDGARD_URL
-      const daemonUrl = getConfig().REACT_APP_THORCHAIN_NODE_URL
-      const thorSwapper = new ThorchainSwapper({
-        daemonUrl,
-        midgardUrl,
-        adapterManager,
-        web3: ethWeb3,
-      })
-      await thorSwapper.initialize()
-      _swapperManager.addSwapper(thorSwapper)
-    })()
+    const midgardUrl = getConfig().REACT_APP_MIDGARD_URL
+    const daemonUrl = getConfig().REACT_APP_THORCHAIN_NODE_URL
+    const thorSwapper = new ThorchainSwapper({
+      daemonUrl,
+      midgardUrl,
+      adapterManager,
+      web3: ethWeb3,
+    })
+    await thorSwapper.initialize()
+    swapperManager.addSwapper(thorSwapper)
   }
 
   if (flags.OsmosisSwap) {
     const osmoUrl = `${getConfig().REACT_APP_OSMOSIS_NODE_URL}/lcd`
     const cosmosUrl = `${getConfig().REACT_APP_COSMOS_NODE_URL}/lcd`
     const osmoSwapper = new OsmosisSwapper({ adapterManager, osmoUrl, cosmosUrl })
-    _swapperManager.addSwapper(osmoSwapper)
+    swapperManager.addSwapper(osmoSwapper)
   }
 
   if (flags.LifiSwap) {
     const lifiSwapper = new LifiSwapper()
     await lifiSwapper.initialize()
-    _swapperManager.addSwapper(lifiSwapper)
+    swapperManager.addSwapper(lifiSwapper)
   }
 
   if (flags.OneInch) {
     const oneInchApiUrl = getConfig().REACT_APP_ONE_INCH_API_URL
     const oneInchSwapper = new OneInchSwapper({ apiUrl: oneInchApiUrl })
-    _swapperManager.addSwapper(oneInchSwapper)
+    swapperManager.addSwapper(oneInchSwapper)
+  }
+
+  return swapperManager
+}
+
+export const getSwapperManager = (flags: FeatureFlags): Promise<SwapperManager> => {
```

Comment:
second part of the fix, dont attempt to return resolved manager singleton, use singleton promise and allow consumers to await initialisation (which the are trying to do already)